src/components/stack/index.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+const Stack = ({
+	additionalClassNames,
+	alignment,
+	children,
+	direction,
+	divider,
+	flexType,
+	justification,
+	gap,
+	wrap,
+}) => {
+	const defaultAndAdditionalClassnames = `c-stack${additionalClassNames ? ` ${additionalClassNames}` : ''}`;
+	return (
+		<div
+			className={defaultAndAdditionalClassnames}
+			data-style-direction={direction}
+			data-style-justification={justification}
+			data-style-alignment={alignment}
+			data-style-flextype={flexType}
+			data-style-wrap={wrap}
+			style={{ '--c-stack-gap': gap }}
+		>
+			{ Array.isArray(children)
+				? children.map((child) => (
+					<Fragment key={child.key}>
+						{ child }
+						{divider && <hr aria-hidden="true" />}
+					</Fragment>
+				)) : (
+					<>
+						{children}
+						{divider && <hr aria-hidden="true" />}
+					</>
```

Comment:
Odd that you'd mix the `Fragment` shorthand (`<></>`) and long form.  Suggest just using the `Fragment` if you are bringing it in.
```suggestion
					</Fragment>
				)) : (
					<Fragment>
						{children}
						{divider && <hr aria-hidden="true" />}
					</Fragment>
```

Would take it one step further and just put your logic inside the fragment code block.

Reply:
May not need it with the suggestions Matthew recommended to handle children iteration using built in React constructs.